### ISSUE TITLE
Suggestion: Make recommendation of seperating DB connection configuration from primary codebase

### ISSUE THREAD

**Issue by kentfredric** [ironcamel/DBICx-Sugar#1]
This module exists for exactly the purpose of allowing machine-specific configuration ( which includes DB connection passwords, etc ) to be isolated from the main code base by allowing the configuration to be referred to in the main code base by an alias: 
- https://metacpan.org/pod/DBIx::Class::Schema::Config

It may be in your interests to document something along those lines, instead of recommending the bad default of hard-coding DB connection strings and passwords in your project code.
- https://metacpan.org/pod/DBIx::Config , by the same author, is also available for  cases where the above is not suitable.

**Comment by ironcamel** [comment]
I'm not sure exactly what you are suggesting.

> ... instead of recommending the bad default of hard-coding DB connection strings and passwords in your project code

I document both the file config and code config approaches.  I don't recommend one approach over the other.  The code config approach is necessary for https://metacpan.org/pod/Dancer::Plugin::DBIC and https://metacpan.org/pod/Dancer2::Plugin::DBIC to be able to use this, which is the original motivation for creating those module.  If you see a usage example it might make more sense: https://github.com/ironcamel/Dancer-Plugin-DBIC/blob/v0.2102/lib/Dancer/Plugin/DBIC.pm

If you have specific suggestions on improving the documenation or anything else, feel free to send a pull request. Thanks.

**Comment by kentfredric** [comment]
> Configuration can be automatically parsed from a `config.yaml` or `config.yml` file in the current working directory, or it can be explicitly set with the config function:
> 
> ``` yaml
> dbicx_sugar:
>  default:
>    dsn: dbi:SQLite:dbname=myapp.db
>    schema_class: MyApp::Schema
> ```

Important part being "current working directory" , which I infer is "In your project root"

The suggestion would be to instead to:

(Important: System/User Global Configuration, not ./ )

``` yaml
# /etc/dbic.yaml
---
MY_DATABASE:
    dsn: "dbi:Pg:host=localhost;database=blog"
    user: "TheDoctor"
    password: "dnoPydoleM"
    TraceLevel: 1
```

And then later

``` yaml
 dbicx_sugar:
  default:
    dsn: MY_DATABASE
    schema_class: MyApp::Schema
```

Completely separating database connection details from database usage details.

`config.yml` can then be checked into git without any concerns about password leakage.

Does the purpose of that make sense to you?

And that technique can be used in both static, and code based connections, the only "evil" part I mention is directly coupling 

a. Where the database is
b. How to authenticate with the database

With a given codebase.

I hope that's clear, but I am rather tired right now. =)

I'd rather of course demonstrate what I'm trying to achieve ( to make sure I'm achieving the right thing ) before I attempt to hack patches in that direction.

**Comment by ironcamel** [comment]
Just to be clear, are you suggesting using 2 config files, a `/etc/dbic.yaml` and a local `./config.yaml` ?

> Does the purpose of that make sense to you?

Yes, I understand what you are trying to achieve. In my projects, separating the db auth from the codebase is already solved because the `config.yaml` is generated by a config management tool, not committed in the codebase. But I am open to making this separation easier for others.

How about something like https://metacpan.org/pod/DBIx::Class::Schema::Config#CONFIG-FILES allowing you to specify the location of the config file via an env variable, and then falling back to checking some default locations for it?

**Comment by kentfredric** [comment]
Ah. Yeah. That sounds workable. As long as the code samples/examples gently encourage the way forward being to avoid the credentials in the primary codebase, I'm all for it, and something like CONFIG_FILES referenced might fit that bill.

My mental consideration is really for people who stumble on DBICx::Sugar, on its own somehow, and then use it as instructions on what to do.

Readers of your document may be unaware of your config management tool, or best practices surrounding things, and DBICx::Sugar could plausibly be one of the first places they see.

My mentality is to have the first and foremost examples given to be "the best way", the way that people should try first if they don't know any better.

At present, a top to bottom reading says
- Configure it in your code directly like this the best option
- Or put it in ./config.yml like this as a second best option

And the current flow inadvertently suggests manually putting the passwords and stuff directly in ./config.yml   is the prescribed use.

**Comment by ironcamel** [comment]
Makes sense. I will update it the config logic and the documentation. Thanks for your suggestions.
